executor/src/witgen/mod.rs

```diff
@@ -299,6 +302,43 @@ impl<'a, 'b, T: FieldElement> WitnessGenerator<'a, 'b, T> {
         }
         witness_cols
     }
+
+    fn range_constraint_multiplicity_witgen(
+        fixed: &FixedData<T>,
+        columns: &mut HashMap<String, Vec<T>>,
+    ) {
+        record_start(RANGE_CONSTRAINT_MULTIPLICITY_WITGEN);
+
+        // Several range constraints might point to the same target
+        let mut multiplicity_columns = BTreeMap::new();
+
+        // Count multiplicities
+        for (source_id, target) in &fixed.global_range_constraints.phantom_range_constraints {
+            let size = fixed.fixed_cols[&target.column]
+                .values
+                .get_uniquely_sized()
+                .unwrap()
+                .len();
+            let multiplicities = multiplicity_columns
+                .entry(target.multiplicity_column)
+                .or_insert_with(|| vec![0; size]);
+            assert_eq!(multiplicities.len(), size);
+            for value in columns.get(fixed.column_name(source_id)).unwrap() {
+                let index = value.to_degree() as usize;
+                multiplicities[index] += 1;
+            }
+        }
+
+        // Convert to field elements and insert into columns
+        for (poly_id, values) in multiplicity_columns {
+            columns.insert(
+                fixed.column_name(&poly_id).to_string(),
+                values.into_iter().map(T::from).collect(),
```

Comment:
Why not already build a vector of T in the first place?

Reply:
Because incrementing `usize` is cheaper, especially on large fields. But this is probably negligible, so I'm happy to change that if you think it would be better.

Reply:
Ok that was my guess. Yeah I have no idea either way so let's keep it like this.